master/pkg/searcher/actions.go

```diff
@@ -0,0 +1,62 @@
+package searcher
+
+import (
+	"fmt"
+
+	"github.com/determined-ai/determined/master/pkg/nprand"
+)
+
+// Action is an action that a searcher would like to perform.
+type Action interface {
+	String() string
```

Comment:
if you want a disciminated union, you can use the same trick i did with `sproto.ResourceEvent`

Reply:
yeah, no typing.Unions here